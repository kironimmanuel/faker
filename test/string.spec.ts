import { afterEach, describe, expect, it } from 'vitest';
import { faker } from '../src';
import type { StringModule } from '../src/modules/string';
import { seededRuns } from './support/seededRuns';

const NON_SEEDED_BASED_RUN = 5;

const functionNames: (keyof StringModule)[] = ['uuid', 'hexadecimal'];

describe('string', () => {
  afterEach(() => {
    faker.locale = 'en';
  });

  for (const seed of seededRuns) {
    describe(`seed: ${seed}`, () => {
      for (const functionName of functionNames) {
        it(`${functionName}()`, () => {
          faker.seed(seed);

          const actual = faker.string[functionName]();

          expect(actual).toMatchSnapshot();
        });
      }

      describe('hexadecimal()', () => {
        it('should return a deterministic hex of given length', () => {
          faker.seed(seed);

          const actual = faker.string.hexadecimal(42);
          expect(actual).toMatchSnapshot();
        });
      });
    });
  }

  // Create and log-back the seed for debug purposes
  faker.seed(Math.ceil(Math.random() * 1_000_000_000));

  describe(`random seeded tests for seed ${JSON.stringify(
    faker.seed()
  )}`, () => {
    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
      describe(`uuid()`, () => {
        it('generates a valid UUID', () => {
          const UUID = faker.string.uuid();
          const RFC4122 =
            /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/;
          expect(UUID).toMatch(RFC4122);
        });
      });

      describe(`hexadecimal()`, () => {
        it('generates single hex character when no additional argument was provided', () => {
          const hex = faker.string.hexadecimal();
          expect(hex).toMatch(/^[0-9a-f]*$/i);
          expect(hex).toHaveLength(1);
        });

        it('generates a random hex string', () => {
          const hex = faker.string.hexadecimal(5);
          expect(hex).toMatch(/^[0-9a-f]*$/i);
          expect(hex).toHaveLength(5);
        });
      });
    }
  });
});
