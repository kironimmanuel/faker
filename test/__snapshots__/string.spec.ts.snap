// Vitest Snapshot v1

exports[`string > random() > should return a deterministic string of given length 1`] = `"Cky2eiXX/J/*&Kq@X.b]\\"&{dnx4!1}2Z=YQ!I#<QYF"`;

exports[`string > random() > should return a deterministic string of given length 2`] = `"9U/4:SK$>6QX9@{:e=+kD)[B,e|/Jqjjj!BLGDWQgC"`;

exports[`string > random() > should return a deterministic string of given length 3`] = `"wKti5-}$_/\`4hHA0afl\\"h^]dnwI<q|p|5KWu3/CZ|J"`;

exports[`string > seed: 42 > alpha() 1`] = `"t"`;

exports[`string > seed: 42 > alphaNumeric() 1`] = `"n"`;

exports[`string > seed: 42 > hexadecimal() > should return a deterministic hex of given length 1`] = `"8BE4ABdd39321aD7d3fe01FfCE404F4d6db0906bd8"`;

exports[`string > seed: 42 > hexadecimal() 1`] = `"8"`;

exports[`string > seed: 42 > random() 1`] = `"Cky2eiXX/J"`;

exports[`string > seed: 42 > uuid() 1`] = `"5cf2bc99-2721-407d-992b-a00fbdf302f2"`;

exports[`string > seed: 1211 > alpha() 1`] = `"W"`;

exports[`string > seed: 1211 > alphaNumeric() 1`] = `"V"`;

exports[`string > seed: 1211 > hexadecimal() > should return a deterministic hex of given length 1`] = `"EaDB42F0e3f4A973fAB0AeefCE96DFCF49cD438dF9"`;

exports[`string > seed: 1211 > hexadecimal() 1`] = `"E"`;

exports[`string > seed: 1211 > random() 1`] = `"wKti5-}$_/"`;

exports[`string > seed: 1211 > uuid() 1`] = `"e7ec32f0-a2a3-4c65-abbd-0caabde64dfd"`;

exports[`string > seed: 1337 > alpha() 1`] = `"n"`;

exports[`string > seed: 1337 > alphaNumeric() 1`] = `"g"`;

exports[`string > seed: 1337 > hexadecimal() > should return a deterministic hex of given length 1`] = `"5c346ba075bd57F5A62B82d72AF39CBBB07a98cbA8"`;

exports[`string > seed: 1337 > hexadecimal() 1`] = `"5"`;

exports[`string > seed: 1337 > random() 1`] = `"9U/4:SK$>6"`;

exports[`string > seed: 1337 > uuid() 1`] = `"48234870-5389-445f-8b41-c61a52bf27dc"`;
